### DISCUSSION TITLE
Netlify CMS (optional)

### DISCUSSION THREAD

**Discussion by ibragimof** [razonyang/hugo-theme-bootstrap#154] [Q&A] [answered]
# Newly in "Hugo themes"

I'm new user in "static websites" topic and "Hugo" so on.. I've some mistakes while building my blog with this theme. There is some questions:

------
### 1. The structure
Other themes ([example](https://github.com/adityatelange/hugo-PaperMod)) creates "branch"es in github, so you can easily switch between "exampleSite" and  direct content of "theme" folder in Hugo.

In this case I've no idea which folder and which file necessary in "theme" folder of "Hugo" and where is "config" file in root and so, and so....

-----

### 2. Online CMS

The online builders such as [Forestry](forestry.io) or [NetlifyCMS](netlifycms.org) are giving so much errors while deploying site with this theme... Can anyone show one or some config files ("front-matter configs" if I understood) for this services?

---------

### 3. Auto updating theme

If I clone this theme to my web-blog (which lives in github as a repository) and deploy with netlify — in this case can I set auto-update for this theme? (is this works via "git*-submodules?)

-------
*P.S. Sorry my english is very bad*

**Comment by ibragimof** [comment]
# UPDATE

No need support, I already have the answers for first two questions.

#### with 3rd question — you are welcome )

**Comment by razonyang** [comment, marked as answer]
> The structure

This repository contains theme's files and the example site's files.

Except the example's files, which path is **exampleSite**, the rest of files are belonged to theme.

> Online CMS

See also [Host on Netlify](https://gohugo.io/hosting-and-deployment/hosting-on-netlify/)

> Auto updating theme

You should install this theme as a submodule of your repository instead of  cloning it.
Once this theme was installed as a submodule, you can upgrade theme via:

```shell
git submodule update --remote --recursive
```

**There is a [blog demo](https://github.com/razonyang/hugo-theme-bootstrap-demo) that hosted on Netlify.** Hope that may help.
